extension/src/Xliff/XLIFFDocument.ts

```diff
@@ -766,27 +758,21 @@ export class Target implements TargetInterface {
     let _textContent = "";
     let _stateValue = null;
     let _stateQualifierValue = undefined;
-    if (!isNullOrUndefined(target)) {
-      _textContent = isNullOrUndefined(target.textContent)
-        ? ""
-        : target.textContent;
-      _stateQualifierValue = target.getAttributeNode("state-qualifier")?.value;
-      _stateQualifierValue = isNullOrUndefined(_stateQualifierValue)
-        ? undefined
-        : _stateQualifierValue.toLowerCase();
-      if (!isNullOrUndefined(target.getAttributeNode("state")?.value)) {
-        _stateValue = isNullOrUndefined(target.getAttributeNode("state")?.value)
-          ? TargetState.new
-          : target.getAttributeNode("state")?.value.toLowerCase();
-      }
+    _textContent = target.textContent ?? "";
+    _stateQualifierValue = target.getAttributeNode("state-qualifier")?.value;
+    _stateQualifierValue = _stateQualifierValue?.toLowerCase() ?? undefined;
+    if (target.getAttributeNode("state")?.value !== undefined) {
+      _stateValue =
+        target.getAttributeNode("state")?.value?.toLowerCase() ??
+        TargetState.new;
     }
+
     const newTarget = new Target(
       _textContent,
-      isNullOrUndefined(_stateValue) ? null : (_stateValue as TargetState)
+      (_stateValue as TargetState) ?? null
```

Comment:
when is `_stateValue` `undefined`?
I believe that it can be `null`, a string or `TargetState.new`
what does a casting `null` to a class return?

Reply:
Good question. I'll need to test that out.

Reply:
Well casting `null as TargetState`  in this case returns `null`. I've added a test for it. See `Cast as TargetState` in `XLIFFTypes.test.ts`. I removed the null coalescing operator from the casting part, as you've deducted it can not be undefined. Both `undefined` and `null` seems to be cast (to `undefined` and `null`)  without error. Good to know!

I would like to change the whole assignment but that can wait, trying to keep the bug count of this low :sweat_smile: